### PR TITLE
Fix CA Timeout and show progress to the user

### PR DESCRIPTION
**Pull request by brimstone**
I know it's not really "progress", but keeping track of the highest difficulty seen so far gives that illusion to the user.

### CODE REVIEW

File: pkg/identity/certificate_authority.go [hunk 1]
```diff
@@ -178,16 +179,20 @@ func (pc PeerCAConfig) Load() (*PeerCertificateAuthority, error) {
...
 		opts.Concurrency = 1
 	}
 	ctx, cancel := context.WithCancel(ctx)
 
+	log.Printf("Generating a certificate matching a difficulty of %d\n", opts.Difficulty)
 	eC := make(chan error)
 	caC := make(chan FullCertificateAuthority, 1)
 	for i := 0; i < int(opts.Concurrency); i++ {
-		go newCAWorker(ctx, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
+		go newCAWorker(ctx, i, &highscore, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
```
Comment (egonelbre) [inline pkg/identity/certificate_authority.go]: Why is this highscore necessary?

File: pkg/identity/utils.go [hunk 2]
```diff
@@ -12,8 +12,10 @@ import (
 	"encoding/asn1"
 	"encoding/pem"
 	"io/ioutil"
+	"log"
```
Comment (egonelbre) [inline pkg/identity/utils.go]: Pass in `zap.Logger` rather than relying on the global logger.

File: pkg/identity/certificate_authority.go (hunk 1 above)
Comment (brimstone) [reply pkg/identity/certificate_authority.go]: It's to give the illusion of progress. It's pass in like this so each worker can know what the current "highscore" is and display the new one to the user.

File: pkg/identity/utils.go (hunk 2 above)
Comment (brimstone) [reply pkg/identity/utils.go]: How? Logging is super weird in this project and I haven't figured it out yet. Is there a good example somewhere?

File: pkg/identity/certificate_authority.go [hunk 3]
```diff
@@ -178,16 +179,20 @@ func (pc PeerCAConfig) Load() (*PeerCertificateAuthority, error) {
...
 	}
 
 	select {
```
Comment (egonelbre) [inline pkg/identity/certificate_authority.go]: Can we just use a for loop with delays and print `.`?

Comment (brimstone) [reply pkg/identity/certificate_authority.go]: That's not nearly as fun, and I've already figured out how to do it this way. ;)

Comment (egonelbre) [reply pkg/identity/certificate_authority.go]:
```
fmt.Print("generating certificate: ")
defer fmt.Println()
for {
	select {
	case <-ticker.C:
		fmt.Print(".")
	case ca := <-caC:
		return &ca, nil
	case err := <-eC:
		return nil, err
	case <-ctx.Done():
		return nil, ctx.Err()
	}
}
```

**Pull request merged**